docs/src/modules/spring/pages/actions.adoc

```diff
@@ -1 +1,73 @@
 = Implementing Actions
+
+include::ROOT:partial$include.adoc[]
+include::partial$actions.adoc[]
+
+In this first example, you will learn how to implement an Action as a pure stateless function. You will create a `FibonacciAction` that takes a number and returns the
+next number in the Fibonacci series.
+
+== Implementing the Action
+
+To implement this action you need the following:
+
+- Extend our class from `kalix.javasdk.action.Action`. This is generic. No matter what action you want to create you always need to extend from link:{attachmentsdir}/testkit/kalix/javasdk/action/Action.html[`Action` {tab-icon}, window="new"]. The rest is specific to this case
+- Add the Spring annotation @RequestMapping to provide a REST endpoint for the function. Here the stateless function should be reachable via HTTP.
+- Add the Spring annotations @GetMapping and @PostMapping to provide paths for GET and POST to calculate the Fibonacci of a number. Both functions do the same thing and implementation-wise the function exposed with GET calls the function exposed with POST.  
+
+
+[source,java,indent=0]
+.src/main/java/com/example/fibonacci/FibonacciAction.java
+----
+include::example$spring-fibonacci-action/src/main/java/com/example/fibonacci/FibonacciAction.java[tag=implementing-action]
+----
+<1> `isFibonacci` checks if a number is a Fibonacci number
+<2> `nextFib` calculates the next number.
+<3> This `nextNumber` implementation calls the `nextNumber` implementation below.
+<4> The `nextNumber` implementation first checks if the input number belongs to the Fibonacci series. If so, it calculates the
+next number and builds a reply using `effects().reply()`.
+<5> Otherwise, if the input number doesn't belong to the Fibonacci series, it builds an `Effect` reply error.
+
+Actions return effects, that is, `Action.Effect`. You have different types of effects. A reply, an error, a forward to the call to another component, and to all of those you can add side effects. Here you want only the result of the calculation or an error. Therefore you are using `.reply` and `.error`.
+
+=== Multiple replies / reply streaming
+
+An Action may return a stream of integers. To do this you need to define the return type as `reactor.core.publisher.Flux<Effect<Integer>>`. 
+
+The stream may publish an arbitrary number of replies.
+
+ifdef::todo[TODO: add a streamed Fib series calculation]
```

Comment:
Shall we open a separate issue for this if we think it's worth? Just so we don't forget it forever?

Reply:
I'm working on this right now. I'll add it to the PR

Reply:
https://github.com/lightbend/kalix/issues/7787